From e948c74332e5385383e8f831de8cfc29d1b2a50e Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Sat, 16 Mar 2024 23:36:01 +0100
Subject: [PATCH] kconf_update: Also run .upd files if times match but done is
 empty

Workaround that kconf_update5 adds entries for .upd files it doesn't run.

BUG: 483795
---
 src/kconf_update/kconf_update.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/kconf_update/kconf_update.cpp b/src/kconf_update/kconf_update.cpp
index 0d1cd251..348e1c9a 100644
--- a/src/kconf_update/kconf_update.cpp
+++ b/src/kconf_update/kconf_update.cpp
@@ -151,9 +151,11 @@ QStringList KonfUpdate::findUpdateFiles(bool dirtyOnly)
             KConfigGroup cg(m_config, fileName);
             const qint64 ctime = cg.readEntry("ctime", 0);
             const qint64 mtime = cg.readEntry("mtime", 0);
+            const QString done = cg.readEntry("done", QString());
             if (!dirtyOnly //
                 || (ctime != 0 && ctime != info.birthTime().toSecsSinceEpoch()) //
-                || mtime != info.lastModified().toSecsSinceEpoch()) {
+                || mtime != info.lastModified().toSecsSinceEpoch() //
+                || (mtime != 0 && done.isEmpty())) {
                 result.append(file);
             }
         }
-- 
2.43.0

